src/Ttl.php

```diff
@@ -0,0 +1,233 @@
+            is_int($ttl) => self::seconds($ttl),
+        };
+    }
+
+    /**
+     * Creates a Ttl object from a DateInterval.
+     *
+     * @param DateInterval $interval The interval to convert to TTL.
+     *
+     * @throws \InvalidArgumentException If the DateInterval results in a negative TTL.
+     *
+     * @return self TTL instance.
+     */
+    public static function fromInterval(DateInterval $interval): self
+    {
+        $seconds = (new DateTime('@0'))
+            ->add($interval)
+            ->getTimestamp();
+
+        return new self($seconds);
+    }
+
+    /**
+     * Create TTL from seconds.
+     *
+     * @param int $seconds Number of seconds.
+     *
+     * @return self TTL instance.
+     */
+    public static function seconds(int $seconds): self
+    {
+        return new self($seconds);
+    }
+
+    /**
+     * Create TTL from minutes.
+     *
+     * @param int $minutes Number of minutes.
+     *
+     * @return self TTL instance.
+     */
+    public static function minutes(int $minutes): self
+    {
+        return new self($minutes * self::SECONDS_IN_MINUTE);
+    }
+
+    /**
+     * Create TTL from hours.
+     *
+     * @param int $hours Number of hours.
+     *
+     * @return self TTL instance.
+     */
+    public static function hours(int $hours): self
+    {
+        return new self($hours * self::SECONDS_IN_HOUR);
+    }
+
+    /**
+     * Create TTL from days.
+     *
+     * @param int $days Number of days.
+     *
+     * @return self TTL instance.
+     */
+    public static function days(int $days): self
+    {
+        return new self($days * self::SECONDS_IN_DAY);
+    }
+
+    /**
+     * Creates a TTL representing "forever" (no expiration).
+     */
+    public static function forever(): self
+    {
+        return new self(null);
+    }
+
+    /**
+     * Checks whether the TTL represents "forever".
+     */
+    public function isForever(): bool
+    {
+        return $this->value === null;
+    }
+
+    /**
+     * Checks whether the TTL is expired.
+     *
+     * @return bool True if TTL <= 0 and not forever
+     */
+    public function isExpired(): bool
+    {
+        return !$this->isForever() && $this->value <= 0;
+    }
+
+    /**
+     * Returns TTL state.
+     */
+    public function status(): int
+    {
+        if ($this->isForever()) {
+            return self::EXPIRATION_FOREVER;
+        }
+
+        if ($this->isExpired()) {
+            return self::EXPIRATION_EXPIRED;
+        }
+
+        return self::EXPIRATION_LIVE;
+    }
+
+    /**
+     * Returns the expiration timestamp relative to $now.
+     *
+     * - For expired TTL, returns EXPIRATION_EXPIRED (-1)
+     * - For forever TTL, returns EXPIRATION_FOREVER (0)
+     * - For live TTL, returns $now + value
+     *
+     * @param int $now Reference timestamp
+     */
+    public function toExpiration(int $now): int
+    {
+        return match ($this->status()) {
```

Comment:
Seems, this is `ArrayCache` internal logic. Expiration "0" or "-1" in common case is strange.